packages/imt.sol/contracts/internal/InternalLeanIMT.sol

```diff
@@ -234,6 +234,8 @@ library InternalLeanIMT {
     ) internal returns (uint256) {
         if (newLeaf >= SNARK_SCALAR_FIELD) {
             revert LeafGreaterThanSnarkScalarField();
+        } else if (oldLeaf == 0) {
+            revert LeafCannotBeZero();
```

Comment:
Hey! Thanks for the idea. I used this error because in that line of code, that leaf can't be 0 if we want to continue with the flow of the algorithm, so if a developer call the function, they will notice that one of the leaves they sent is 0.

I will update the implementation with this idea: https://github.com/privacy-scaling-explorations/zk-kit/issues/228#issuecomment-2016928091 posted by @0xbok